build-id-versioning/workflows.go

```diff
@@ -0,0 +1,134 @@
+package build_id_versioning
+
+import (
+	"context"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/workflow"
+	"time"
+)
+
+// SampleChangingWorkflowV1 is a workflow we'll be making changes to, and represents the first
+// version
+func SampleChangingWorkflowV1(ctx workflow.Context) error {
+	workflow.GetLogger(ctx).Info("Changing workflow v1 started.", "StartTime", workflow.Now(ctx))
+
+	// This workflow will listen for signals from our starter, and upon each signal either run
+	// an activity, or conclude execution.
+	signalChan := workflow.GetSignalChannel(ctx, "do-next-signal")
+	for {
+		var signal string
+		signalChan.Receive(ctx, &signal)
+
+		if signal == "do-activity" {
+			workflow.GetLogger(ctx).Info("Changing workflow v1 running activity")
+			ao := workflow.ActivityOptions{
+				StartToCloseTimeout: 10 * time.Second,
+			}
+			ctx1 := workflow.WithActivityOptions(ctx, ao)
```

Comment:
Feel free to move out of loop near top if you want (or at least use the created context to also `Get`, though not required is often best practice). Will definitely cut down on repeated code in your next workflow.

Reply:
I think it's worth keeping just to preserve the idea that you want the exact same code in the old branch.